staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go

```diff
@@ -273,6 +273,12 @@ func Start(ctx context.Context, nodeServer interface{}, opts ...Option) (result
 	if o.draAddress == "" {
 		return nil, errors.New("DRA address must be set")
 	}
+	if o.kubeClient == nil {
+		return nil, errors.New("kubeclient must be set")
+	}
+	if o.nodeName == "" {
+		return nil, errors.New("node name must be set")
+	}
```

Comment:
Is this always required? Honest question.

Reply:
Without this it is impossible to publish a resource slice. Unless there are other use cases planned for future than DRA drivers publishing ResourceSlices - kubeclient is always required. Reading the interface - publishing the ResourceSlice is the only purpose of DRAPlugin.

As for the `nodeName` - something needs to own the resource slice, and also nodeName is used as the name of the pool in `resourceSlice.DriverResources.Pools` map. I believe empty value works for map, but then all the ResourceSlices will be in the same pool.

For the Owner reference - nodeUID is another option, but it is optional because the `ResourceSliceController` will look it up based on nodeName. So nodeName appears to be needed according to my observations.

Reply:
> Without this it is impossible to publish a resource slice.

Not all kubelet plugins need to do that (classic DRA, network-attached devices).

Let's add an error return to `PublishResources` and return an error there if it gets called without preparing for it. The documentation also should be improved.